src/converse-autocomplete.js

```diff
@@ -147,7 +148,7 @@ export class AutoComplete {
             'auto_first': false, // Should the first element be automatically selected?
             'data': a => a,
             'filter': FILTER_CONTAINS,
-            'sort': config.sort === false ? false : SORT_BYLENGTH,
+            'sort': config.sort === false ? false : SORT_BY_QUERY,
```

Comment:
`SORT_BY_QUERY` is not a very descriptive name  for what the function is doing.

I instead suggest `SORT_BY_QUERY_POSITION`